### PR TITLE
Ignore notification close event & clear notifications when marking accounts unread chat as noticed

### LINKED ISSUE #4002: "Mark All as Read" does not remove notifications
the gist of  "Mark All as Read" is to get chats out of ones way. therefore, and for consistency, also notifications should be removed when things are marked as read.

this issue may be one cause of stale notifications of https://github.com/deltachat/deltachat-desktop/issues/3991 , however, i doubt it is the only one as i was not using "Mark All as Read" until recently (i was not aware it exist :)

cmp https://github.com/deltachat/deltachat-desktop/issues/3979

### PR DESCRIPTION
**Pull request by nicodh**
fixes #4002

### CODE REVIEW

**Comment by nicodh** [comment]
I will test on Windows to resolve #3982

**Comment by nicodh** [comment]
I also found an entry in the ChatListItemArchiveLink context menu "Mark al as read" which makes no sense imho since new messages in archived chats are not displayed as new at all.
@Simon-Laux is there a case where this makes sense? Otherwise I will remove it
ArchiveLink-Context-Menu

**Comment by Simon-Laux** [comment]
mute a chat then archive it. then receive a message for it. archived chats show a badge counter that tells the number of archived muted chats with unread messages.
muted chats stay archived when a new message is received.
So this context menu is intentional and useful.

File: src/main/notifications.ts [hunk 1]
```diff
@@ -79,8 +79,8 @@ function showNotification(_event: IpcMainInvokeEvent, data: DcNotification) {
       notify.close()
     })
     notify.on('close', () => {
-      notifications[chatId] =
-        notifications[chatId]?.filter(n => n !== notify) || []
+      // this is triggered when the message is moved to notification center
+      // so we don't remove the instance from the list
```
Comment (Simon-Laux) [inline src/main/notifications.ts:83]: Electron docs says:

> Emitted when the notification is closed by manual intervention from the user.
> ~ https://www.electronjs.org/docs/latest/api/notification#event-close

So I think it makes sense to mention in the comment that this is a bug.

**Comment by nicodh** [comment]
Tested on Windows, Mac & Linux

Works as expected, except a bug in Linux which is not related to deltachat (notifications are never removed from notification)
#/3982 ist not solved with this MR but not really related

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@
 - Fix the problem of Quit menu item on WebXDC apps closes the whole DC app #3995
 - minor performance improvements #3981
 - fix chat list items (e.g. Archive) and contacts not showing up sometimes #4004
+- fix bug notifications not being removed on Mac  #4010
```
Comment (Simon-Laux) [inline CHANGELOG.md:29]: you forgot to mention the fix for clear account notifications

**Pull request merged**